JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt

```diff
@@ -91,83 +93,113 @@ enum class Sections(@StringRes val titleResId: Int) {
 class TabContent(val section: Sections, val content: @Composable () -> Unit)
 
 /**
- * Displays the Interests screen.
+ * Stateful composable that displays the Navigation route for the Interests screen.
  *
  * @param interestsViewModel ViewModel that handles the business logic of this screen
+ * @param showNavRail (state) whether the Drawer or NavigationRail needs to be shown
+ * @param navigateToHome (event) request navigation to Home screen
  * @param openDrawer (event) request opening the app drawer
  * @param scaffoldState (state) state for screen Scaffold
  */
 @Composable
-fun InterestsScreen(
+fun InterestsRoute(
     interestsViewModel: InterestsViewModel,
+    showNavRail: Boolean,
+    navigateToHome: () -> Unit,
     openDrawer: () -> Unit,
     scaffoldState: ScaffoldState = rememberScaffoldState()
 ) {
-    // UiState of the InterestsScreen
-    val uiState by interestsViewModel.uiState.collectAsState()
-
-    // Describe the screen sections here since each section needs 2 states and 1 event.
-    // Pass them to the stateless InterestsScreen using a tabContent.
-    val topicsSection = TabContent(Sections.Topics) {
-        val selectedTopics by interestsViewModel.selectedTopics.collectAsState()
-        TopicList(
-            topics = uiState.topics,
-            selectedTopics = selectedTopics,
-            onTopicSelect = { interestsViewModel.toggleTopicSelection(it) }
-        )
-    }
-
-    val peopleSection = TabContent(Sections.People) {
-        val selectedPeople by interestsViewModel.selectedPeople.collectAsState()
-        PeopleList(
-            people = uiState.people,
-            selectedPeople = selectedPeople,
-            onPersonSelect = { interestsViewModel.togglePersonSelected(it) }
-        )
-    }
-
-    val publicationSection = TabContent(Sections.Publications) {
-        val selectedPublications by interestsViewModel.selectedPublications.collectAsState()
-        PublicationList(
-            publications = uiState.publications,
-            selectedPublications = selectedPublications,
-            onPublicationSelect = { interestsViewModel.togglePublicationSelected(it) }
-        )
+    val tabContent = rememberTapContent(interestsViewModel)
+    val (currentSection, updateSection) = rememberSaveable {
```

Comment:
What is the purpose of this? If I go to the people tab, press home and back to interests, I see topics tab.

Reply:
This works currently for configuration changes on this screen. But it's true that it's not preserved when going to a different screen. Probably we can tackle this in a separate PR. As discussed offline, this could be hoisted in `JetnewsNavGraph`, for example, or in the backStackEntry of the destination